### PR TITLE
Mbst 17344/migrate to applications service client

### PR DESCRIPTION
**Pull request by spraxxy**
Removed references to current system of Application and ApplicationConfiguration and replaced them with ApplicationsServiceClient

### CODE REVIEW

File: src/main/java/org/atlasapi/content/criteria/ContentQueryBuilder.java [hunk 1]
```diff
@@ -30,6 +32,7 @@
 
 	private final List<AtomicQuery> queries;
 	private Selection selection = Selection.ALL;
+	private Optional<Application> application;
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/content/criteria/ContentQueryBuilder.java]: Why do we need to add `application` to the `ContentQuery`? Which code needs this?

File: src/main/java/org/atlasapi/media/entity/simple/ContentIdentifier.java [hunk 2]
```diff
@@ -223,7 +223,7 @@ private static ContentIdentifier create(EntityType type, String uri, String id,
         case SERIES:
             return new SeriesIdentifier(uri, seriesNumber, id);
         default:
-            throw new RuntimeException("Can't create content identifier for " + uri);	    
+            throw new RuntimeException("Can't createDefault content identifier for " + uri);
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/media/entity/simple/ContentIdentifier.java]: Refactor error I think

File: src/main/java/org/atlasapi/media/segment/Segment.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
     public SegmentRef toRef() {
         return new SegmentRef(checkNotNull(this.getId(),
-                "Can't create reference for segment without ID"));
+                "Can't createDefault reference for segment without ID"));
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/media/segment/Segment.java]: Ditto

**Comment by Dalamar42** [comment]
This needs a quick re-review after the merge conflicts are resolved.

File: src/main/java/org/atlasapi/content/criteria/ContentQueryBuilder.java (hunk 1 above)
Comment (jamiesprax) [inline src/main/java/org/atlasapi/content/criteria/ContentQueryBuilder.java]: ContentResolvingSearcher:104

**Comment by jamiesprax** [comment]
The merge conflicts were just imports so no need to re-review

**Pull request closed without merging**